Shell/EqualityProxyMono.cpp

```diff
@@ -273,23 +270,20 @@ Clause* EqualityProxyMono::apply(Clause* cl)
   Clause* res;
   ASS(proxyPremises.isNonEmpty());
   if (proxyPremises.size() == 1) {
-    res = new(clen) Clause(clen,
+    res = Clause::fromStack(*resLits,
         NonspecificInference2(InferenceRule::EQUALITY_PROXY_REPLACEMENT, cl, proxyPremises.top()));
   }
   else {
     UnitList* prems = 0;
     UnitList::pushFromIterator(UnitStack::ConstIterator(proxyPremises),prems);
     UnitList::push(cl,prems);
 
-    res = new(clen) Clause(clen,
+    res = Clause::fromStack(*resLits,
         NonspecificInferenceMany(InferenceRule::EQUALITY_PROXY_REPLACEMENT, prems));
   }
+  // TODO isn't this done atomatically
   res->setAge(cl->age()); // MS: this seems useless; as long as EqualityProxy is only operating as a part of preprocessing, age is going to 0 anyway
```

Comment:
OK, so here we would also get the right thing with taking the age from the main premise.